packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ModelResource.java

```diff
@@ -244,4 +249,84 @@ public Response saveAsNewModel(
+		PetriNetModel petriNetModel = new PetriNetModel();
+		List<PetriNetState> petriNetStates;
+		List<PetriNetTransition> petriNetTransitions = new ArrayList<>();
+		String[] splitStateNames = states != null ? states.split(",") : new String[]{""};
+		if (splitStateNames.length < 1) {
+			return Response.status(Response.Status.BAD_REQUEST).build();
+		}
+		for (String stateName : splitStateNames) {
+			if (stateName.contains("_")) {
+				return Response.status(Response.Status.BAD_REQUEST).build();
+			}
+		}
+		petriNetStates = IntStream.range(0, splitStateNames.length).mapToObj(i -> {
+			PetriNetState petriNetState = new PetriNetState();
+			petriNetState.setId('L'+ Integer.toString(i + 1));
+			petriNetState.setName(splitStateNames[i]);
+			return petriNetState;
+		}).collect(Collectors.toList());
+		for (int i = 0; i < petriNetStates.size(); i++) {
+			for (int j = 0; j < petriNetStates.size(); j++) {
+				if (i != j) {
+					PetriNetTransition petriNetTransition = new PetriNetTransition();
+					petriNetTransition.setId('t' + Integer.toString(i + 1) + (j + 1));
+					petriNetTransition.setInput(Arrays.asList(petriNetStates.get(i).getId()));
+					petriNetTransition.setOutput(Arrays.asList(petriNetStates.get(j).getId()));
+					petriNetTransitions.add(petriNetTransition);
+				}
+			}
+		}
+		petriNetModel.setTransitions(petriNetTransitions);
+		petriNetModel.setStates(petriNetStates);
+		return Response
+			.status(Response.Status.OK)
+			.entity(petriNetModel)
+			.build();
+	}
+
+	@PUT
+	@Path("/generate-age-strata-model")
+	public Response generateAgeStrataModel(
+		@QueryParam("states") final String states) {
+		PetriNetModel petriNetModel = new PetriNetModel();
+		List<PetriNetState> petriNetStates;
+		List<PetriNetTransition> petriNetTransitions = new ArrayList<>();
+		String[] splitStateNames = states != null ? states.split(",") : new String[]{""};
+		if (splitStateNames.length < 1) {
+			return Response.status(Response.Status.BAD_REQUEST).build();
+		}
+		for (String stateName : splitStateNames) {
+			if (stateName.contains("_")) {
+				return Response.status(Response.Status.BAD_REQUEST).build();
+			}
+		}
+		petriNetStates = IntStream.range(0, splitStateNames.length).mapToObj(i -> {
+				PetriNetState petriNetState = new PetriNetState();
+				petriNetState.setId('A'+ Integer.toString(i + 1));
+				petriNetState.setName(splitStateNames[i]);
+				return petriNetState;
+			}).collect(Collectors.toList());
+		for (int i = 0; i < petriNetStates.size(); i++) {
+			for (int j = 0; j < petriNetStates.size(); j++){
+				PetriNetTransition petriNetTransition = new PetriNetTransition();
+				petriNetTransition.setId('c' + Integer.toString(i + 1) + (j+1));
+				petriNetTransition.setInput(Arrays.asList(petriNetStates.get(i).getId(), petriNetStates.get(j).getId()));
+				petriNetTransition.setOutput(Arrays.asList(petriNetStates.get(i).getId(), petriNetStates.get(j).getId()));
+				petriNetTransitions.add(petriNetTransition);
```

Comment:
This algorithm doesn't look right. Can you provide what the output of a 3-age (young, medium, old) looks like?

Also how is the age stratification differ than the location stratification? They look the same.

Reply:
> Why is this written in Java and requires an API call?
> 
> It seems to make more sense if we just generate it on the front end no? Since we are not doing anything with any side-effects?

I was just trying to keep application logic out of the front end; what do you think?